library/alloc/src/boxed.rs

```diff
@@ -2080,6 +2121,48 @@ impl<I> FromIterator<I> for Box<[I]> {
     }
 }
 
+/// `Box` is fundamental, so coherence needs help understanding these impls are okay.
+#[stable(feature = "boxed_slice_into_iter", since = "CURRENT_RUSTC_VERSION")]
+impl<I, A: Allocator> !Iterator for Box<[I], A> {}
+
+/// `Box` is fundamental, so coherence needs help understanding these impls are okay.
+#[stable(feature = "boxed_slice_into_iter", since = "CURRENT_RUSTC_VERSION")]
+impl<'a, I, A: Allocator> !Iterator for &'a Box<[I], A> {}
+
+/// `Box` is fundamental, so coherence needs help understanding these impls are okay.
+#[stable(feature = "boxed_slice_into_iter", since = "CURRENT_RUSTC_VERSION")]
+impl<'a, I, A: Allocator> !Iterator for &'a mut Box<[I], A> {}
```

Comment:
Those are user facing docs, maybe we should write something that can be understood by normal users? (this can be a follow up)

Reply:
fff I thought I fixed these. I'll copy them over from the other PR for `Box<[T; N]>` where I think did.